Adding Popup Helper Text to Add Project Update Form

zhongh opened the pull request:
Initial commit. Message box boundary still not working.

zhongh committed:
Blue popup logo with popup message working for associated publications; message box still missing

zhongh commented:
<img width="790" alt="screen shot 2016-02-05 at 7 15 00 pm" src="https://cloud.githubusercontent.com/assets/11859381/12862772/db7c86f4-cc3c-11e5-82e7-c6eae121e020.png">

mrpatrickwest commented:
I would like to have helper text for all of the fields on that page.

Can we add black border to the popup?

zhongh commented:
I will add helper text for all.
The border is the pain right now. Just can't get the border working but I will keep trying...

mrpatrickwest commented:
Looks like you are not up to date with master. You're missing the instrument reference and the modification label is still there.

zhongh committed:
minor updates regarding the last commit

zednis committed:
Merge branch 'master' into add-help-text-to-add-project-update-form

resolved conflicts on:
#	productMods/i18n/all.properties
#	productMods/templates/freemarker/edit/forms/addProjectUpdate.ftl

zednis commented:
I used Intelli-J IDE to attempt to resolve merge conflicts, but based on the Travis-CI failure something did not work right.

zednis commented:
@mrpatrickwest `ant test` of this branch works on my local machine.  I am not sure if the travis-CI failure was not just a network problem...

Could you checkout a clean repo and test the build of this branch?

am-e commented:
I may have made some additional changes when working on the "refers to instrument" branch.

zhongh committed:
conflict fixed; travis CI test should pass

mrpatrickwest commented:
@am-e May have made changes to ???

mrpatrickwest commented:
There's a commit that changes the submodule versions, that's what's breaking

zhongh commented:
@mrpatrickwest While making some changes to instruments, @am-e also made a little update to the publications part above it. After merging there emerged a redundant line. The commit just now was to fix that conflict.

And how to fix the submodule version problem?

mrpatrickwest commented:
@zednis I did a fresh clone and checkout of the branch on my laptop
git submodule init - works fine
git submodule update - fails for vitro

zhongh committed:
vitro versioning fix

zhongh commented:
Here is how I fixed it:
`git submodule deinit -f .`
`git checkout 183512fd148c935a9dcf83b5aabd99224f203c21 -- vitro`
`git submodule init`
`git submodule update`

am-e commented:
I did make changes but they were all limited to the addProjectUpdateGenerator.java and the ftl, (also the preprocessor), took out some unused variables and added error checking for missing values.

Commit:
had to override border styles to get helper border

The border properties were already defined in the style scope for
HelperInfoHover so had to override them all to get a border around the
helper text.

mrpatrickwest commented on productMods/templates/freemarker/edit/forms/addProjectUpdate.ftl:
```diff
@@ -180,6 +185,39 @@ Set this flag on the input acUriReceiver where you would like this behavior to o
+        background-color:#FFFFFF;
+        padding: 5px;
+        border-radius: 4px;
+        border-top-color: #5C85D6;
```
these border styles were already defined by parent styles so needed to override them.

zhongh replied:
:+1:

mrpatrickwest commented:
need to add helper text for all of the other fields as well. Just add placeholder text in there for now and I'll work with Josh on some good text to put there.

zhongh committed:
All helper texts added for further discussion

zhongh committed:
border style overrides for popup box

zhonghao1014 commented:
All initial placeholder helper texts created for further discussion between Patrick and Josh. Border style overriding for popup box still not working but I will keep trying.
![image](https://cloud.githubusercontent.com/assets/11859381/12989640/29ad31e8-d0d4-11e5-9307-2eb45f045022.png)

zhongh committed:
Fixed small mistake in css scripts and borders working now

zhongh commented:
1) Borders working (Thanks @mrpatrickwest !!)
2) All initial helper texts added for Patrick and Josh to review. 
3) Managed to did a git pull on the local branch on udco, but failed to compile due to lack of sudoer permission.
![image](https://cloud.githubusercontent.com/assets/11859381/12990258/76c6cb3a-d0d7-11e5-93f9-f3cc0b413207.png)

Commit:
Updated helper text per discussions with Josh Wood

Working with Josh Wood from URI, changed text and added a Note below the labels for publications and datasets, and instruments

Commit:
Merge branch 'master' into add-help-text-to-add-project-update-form

Commit:
Merge remote-tracking branch 'origin' into add-help-text-to-add-project-update-form

mrpatrickwest commented:
@zednis @zhonghao1014 @am-e 
Ready to be reviewed. Login, find a project, click the Update tab, click the + to add a project update.

Commit:
Put instrument note in the right place

Like the publication note, needed to put the instrument note in its own div. This way the information i is next to the label and not the note.

am-e commented:
Logged in, checked the help texts, created project update, all seems to be working.

zhongh commented:
:+1: for me

zednis commented:
:+1:
